src/SourceLocator/Type/Composer/Factory/MakeLocatorForComposerJsonAndInstalledJson.php

```diff
@@ -38,19 +38,22 @@ public function __invoke(string $installationPath, Locator $astLocator): SourceL
             throw InvalidProjectDirectory::atPath($installationPath);
         }
 
-        $composerJsonPath  = $realInstallationPath . '/composer.json';
-        $installedJsonPath = $realInstallationPath . '/vendor/composer/installed.json';
+        $composerJsonPath = $realInstallationPath . '/composer.json';
 
         if (! file_exists($composerJsonPath)) {
             throw MissingComposerJson::inProjectPath($installationPath);
         }
 
+        /** @var array{autoload: array{classmap: array<int, string>, files: array<int, string>, psr-4: array<string, array<int, string>>, psr-0: array<string, array<int, string>>}, config: array{vendor-dir: array<string, array<int, string>>}}|null $composer */
+        $composer  = json_decode((string) file_get_contents($composerJsonPath), true);
+        $vendorDir = $composer['config']['vendor-dir'] ?? 'vendor';
+
+        $installedJsonPath = $realInstallationPath . '/' . $vendorDir . '/composer/installed.json';
```

Comment:
We probably need test for these cases:

- `composer.json` with a custom `vendor-dir` specified
- `composer.json` with no `vendor-dir` specified (I'm assuming this is already tested, but I honestly don't remember without looking)

Reply:
I checked, yes it is tested already (also, see https://github.com/Roave/BetterReflection/pull/711#issuecomment-743718259 for testing suggestions!)

Reply:
The `with no vendor-dir specified` test is already there. I'll add the `with custom vendor-dir` test case too.